### PR TITLE
Feature/ch announce in random slot next round

### PR DESCRIPTION
**Pull request by Tejpbit**

### CODE REVIEW

**Comment by Tejpbit** [comment]
[Announce yourself as CH in a random slot instead of always at the end of the round (maybe in the first half of slots or something)](https://trello.com/c/HxsnxUr0/225-announce-yourself-as-ch-in-a-random-slot-instead-of-always-at-the-end-of-the-round-maybe-in-the-first-half-of-slots-or-something)

File: a2-synchrotron-contiki/core/net/mac/chaos/service/cluster/cluster.c [hunk 1]
```diff
@@ -148,7 +166,21 @@ static chaos_state_t process(uint16_t round_count, uint16_t slot,
...
         got_valid_rx = 1;
-        next_state = CHAOS_TX;
+        set_next_state(&next_state, CHAOS_TX);
+    }
+
+
+    if (tentativeAnnouncementSlot != -1) {
+        uint8_t demote = another_announced_CH_noticed(cluster_rx->cluster_head_list, cluster_rx->cluster_head_count);
+        if (demote) {
+            COOJA_DEBUG_PRINTF("Found another CH, demotion")
```
Comment (mattiasnilsen) [inline a2-synchrotron-contiki/core/net/mac/chaos/service/cluster/cluster.c]: remove print

File: a2-synchrotron-contiki/core/net/mac/chaos/service/cluster/cluster.c [hunk 2]
```diff
@@ -148,7 +166,21 @@ static chaos_state_t process(uint16_t round_count, uint16_t slot,
...
+            tentativeAnnouncementSlot = -1; // Demote
+        } else if (slot >= tentativeAnnouncementSlot) {
+            tentativeAnnouncementSlot = -1;
+            cluster_head_state = TENTATIVE;
+            COOJA_DEBUG_PRINTF("cluster, I AM TENTATIVE, slot %d should be tx\n", slot+1);
```
Comment (mattiasnilsen) [inline a2-synchrotron-contiki/core/net/mac/chaos/service/cluster/cluster.c]: same

File: a2-synchrotron-contiki/core/net/mac/chaos/service/cluster/cluster.c [hunk 3]
```diff
@@ -438,8 +471,8 @@ static void heed_repeat(const cluster_head_information_t* cluster_head_list, uin
         uint32_t precision = 1000;
         uint32_t probability = current_CH_prob * precision;
         if(chaos_random_generator_fast() % precision <= probability) {
-            cluster_head_state = TENTATIVE;
-            COOJA_DEBUG_PRINTF("cluster, I AM TENTATIVE\n");
+            tentativeAnnouncementSlot = chaos_random_generator_fast_range(0, CLUSTER_ROUND_MAX_SLOTS/2);
+            COOJA_DEBUG_PRINTF("cluster, I AM TENTATIVE, decided to become CH in slot %d\n", tentativeAnnouncementSlot);
```
Comment (mattiasnilsen) [inline a2-synchrotron-contiki/core/net/mac/chaos/service/cluster/cluster.c:473]: Remove print

**Review by mattiasnilsen — APPROVED** [review]

**Pull request merged**